app/services/site_setting/update.rb

```diff
@@ -3,42 +3,49 @@
 class SiteSetting::Update
   include Service::Base
 
-  options { attribute :allow_changing_hidden, :boolean, default: false }
+  options { attribute :allow_changing_hidden, :array, default: [] }
 
   policy :current_user_is_admin
 
   params do
-    attribute :setting_name
-    attribute :new_value
+    attribute :settings
 
     before_validation do
-      self.setting_name = setting_name&.to_sym
-      self.new_value = new_value.to_s.strip
+      self.settings = self.settings.to_a.map { |key, value| [key.to_sym, value.to_s.strip] }.to_h
     end
 
-    validates :setting_name, presence: true
+    validates :settings, presence: true
 
     after_validation do
-      next if setting_name.blank?
-      self.new_value =
-        case SiteSetting.type_supervisor.get_type(setting_name)
-        when :integer
-          new_value.tr("^-0-9", "").to_i
-        when :file_size_restriction
-          new_value.tr("^0-9", "").to_i
-        when :uploaded_image_list
-          new_value.blank? ? "" : Upload.get_from_urls(new_value.split("|")).to_a
-        when :upload
-          Upload.get_from_url(new_value) || ""
-        else
-          new_value
-        end
+      self.settings =
+        self
+          .settings
+          .to_a
```

Comment:
`settings` should be a hash here because of the `before_validation` block, so a call to `#to_a` shouldn’t be necessary.